pkg/nfd-master/nfd-master.go

```diff
@@ -92,18 +93,26 @@ type NfdMaster interface {
 }
 
 type nfdMaster struct {
-	args      Args
-	nodeName  string
-	server    *grpc.Server
-	ready     chan bool
-	apihelper apihelper.APIHelpers
+	args         Args
+	nodeName     string
+	annotationNs string
+	server       *grpc.Server
+	ready        chan bool
+	apihelper    apihelper.APIHelpers
 }
 
 // Create new NfdMaster server instance.
 func NewNfdMaster(args Args) (NfdMaster, error) {
 	nfd := &nfdMaster{args: args,
 		nodeName: os.Getenv("NODE_NAME"),
-		ready:    make(chan bool, 1)}
+		ready:    make(chan bool, 1),
+	}
+
+	if args.Instance == "" {
+		nfd.annotationNs = AnnotationNsBase
+	} else {
+		nfd.annotationNs = args.Instance + "." + AnnotationNsBase
```

Comment:
nit:
should you use strings.join() like in other places in this file ?

Reply:
The other places have a slice to join. I originally used strings.Join but decided chaged it. Here we only have two values so IMHO this is more readable and practical than doing
```
strings.Join([args.Instance, AnnotationNsBase], ".")
```

WDYT?

Reply:
Make sense, just wanted to point it out and i see you already considered it when coding.